AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt

```diff
@@ -72,7 +73,12 @@ fun DeckPicker.handleNewSync(
             when (conflict) {
                 Connection.ConflictResolution.FULL_DOWNLOAD -> handleDownload(deckPicker, auth, syncMedia)
                 Connection.ConflictResolution.FULL_UPLOAD -> handleUpload(deckPicker, auth, syncMedia)
-                null -> handleNormalSync(deckPicker, auth, syncMedia)
+                null -> {
+                    try { handleNormalSync(deckPicker, auth, syncMedia) } catch (exc: Exception) {
+                        showSnackbar(R.string.hostname_exception)
```

Comment:
We have an existing string already that is effectively the same: https://github.com/ankidroid/Anki-Android/blob/13fff3f24149a1a96658ed168b2380d7bcf2db9e/AnkiDroid/src/main/res/values/02-strings.xml#L381

```suggestion
                        showSnackbar(R.string.check_network)
```